### PR TITLE
[MB-7020][MB-7021] Fix address fieldset component

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR started as just a story to create a reusable component for the Service Member mailing address fields, but as I started creating an example of that in Storybook I realized there were issues with how the `AddressFields` were using both Formik's `Field` component _and_ the MilMove `TextInput` (which uses Formik's `useField` hook). This was resulting in some very confusing and broken code, so I used this branch to clean up & add clarification to the various ways to use `TextInput` and some related components.

To complete the original story AC, I added examples of how to use the existing `AddressFields` component to render the service member address fields (see **Components / Fieldsets / AddressFields / Current Residential Address, With Additional Text** in Storybook). Since this requires minimal code on top of AddressFields, I don't think there's any reason to create a new component for it.

As of this PR, if you want to add a text input to a Formik form you can use the `TextField` component, which is nearly identical to the formerly named `TextInput` component. Since this component uses `useField`, you should _not_ pass a `value` prop to it - this will conflict with the `value` prop that Formik will provide.

- I renamed `TextInput` to `TextField` and simplified the component code. It renders ReactUSWDS's TextInput inside a FormGroup with a label, hint, error, warning as it did before. I also removed the separate `TextInputMinimal` component because it was never actually used outside of `TextField`, and was also duplicating `useField`.
- I moved `TextInputMasked` to its own file and renamed to `MaskedTextField`. I made no code changes.
- I applied changes to both `AddressFields` and `ContactInfoFields`, and the only place in the application where these are used is the `MtoShipmentForm` so I made some required changes there as well. Changes were removing the `values` prop (since this will be passed in via `useField`) and removing the use of Formik's `<Field />` component.
- I added some tests and also converted some existing tests to use RTL instead of Enzyme

## Reviewer Notes

For the story AC, verify the Current Residential Address Storybook example matches the "Current mailing address" block, and the With Additional Text example matches the "Backup mailing address" block in the Zeplin design ([desktop](https://zpl.io/29X40Pd)) ([mobile](https://zpl.io/b6XZ7mg))

There are no expected user-facing changes, but it's worth spot checking forms that use the changed components:
- Customer / MTO Shipment Form (add/edit shipments)
- Office / Allowances Detail Form
- Office / Orders Detail Form
- Office / Reject Service Item modal

## Setup

To run the app locally:
```sh
make server_run
make client_run
```

You can also view the AddressFields component in Storybook:
```
yarn storybook
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [MB-7020](https://dp3.atlassian.net/browse/MB-7020) & [MB-7021](https://dp3.atlassian.net/browse/MB-7021)

## Screenshots

image

image

image

image

### CODE REVIEW

**Review by lynzt — APPROVED** [review]
LGTM 🚢

File: src/components/form/fields/MaskedTextField.test.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
```
[the remark this answers is not in the document]
Comment (suzubara) [reply src/components/form/fields/MaskedTextField.test.jsx:2]: I was just renaming the file without making any code changes

**Review by amanda5iu — COMMENTED** [review]
This looks great Suz! Can we try your suggestion to add padding between the fields so that they aren't squished in the error state? I think it's ok if adjacent fields are misaligned for a little bit.

**Comment by Shkeating** [comment]
from a design standpoint, only thing I see off is the space above the signature on the `Now for the official part`field, which should be 24px as per [the designs on Zeplin](https://zpl.io/VDeo0xl): 
image

**Review by YanZ777 — APPROVED** [review]
From a code standpoint, the changes look good to me!

**Comment by suzubara** [comment]
@amanda5iu @Shkeating Ok, I think both issues have been resolved! The margin in the Agreement form should be fixed: 
image

and I updated the error styles so they're in line with the USWDS defaults (we were overwriting a margin that I don't think we needed to). I also added a Storybook example called "Inside Section Wrapper" to show how the form would look when on the page: 
image

**Review by Shkeating — APPROVED** [review]
changes look good!! 😎✅

**Comment by Shkeating** [comment]
also, hooray to all the pages getting tracked in storybook!! 🥳

I did notice on your Kitchen sink form page, that this checkbox could use some space above it.

image

**Review by amanda5iu — APPROVED** [review]
Looks so good! 🎉

**Pull request merged**